src/components/Inputs/validation.ts

```diff
@@ -0,0 +1,165 @@
+import * as yup from 'yup';
+import { SchemaOf, ValidationError } from 'yup';
+import {
+  InputModGPO,
+  InputModCHP,
+  InputModGP
+} from '@ucdavis/tea/out/models/input.model';
+import { FrcsInputs, TechnoeconomicModels } from '../../models/Types';
+
+// FrcsInputs
+let frcsSchema: SchemaOf<FrcsInputs> = yup.object().shape({
+  system: yup.string().required(),
+  dieselFuelPrice: yup.number().required().min(0),
+  treatmentid: yup.number().required().min(0)
+});
+
+export const checkFrcsValidity = async (inputs: FrcsInputs) => {
+  try {
+    await frcsSchema.validate(inputs);
+  } catch (err) {
+    if (err instanceof ValidationError) {
+      return err.errors;
+    }
+  }
+
+  return [];
+};
+
+// Tea Example
+// TODO: each schema need to be defined as SchemaOf<InputModGPO> or correct type
+// Then each of the properties need to be listed with proper validation rules.
+// Eventually we'll even want to make sure the values are between certain numbers, but for now
+// we'll just make sure that any percentages are between 0 and 100 and leave everything else alone.
+let capitalCostElements = yup.object().shape({
```

Comment:
It might be overkill but could you add the schema types to these?  That'll help us make sure we didn't miss any.  So for example this line would become: `let capitalCostElements: SchemaOf<CapitalCostInputModGP> = yup.object().shape({`

Reply:
Yea I can add that in